### ISSUE TITLE
[Question] Support for running test in distributed mode, something like selenium hub/grid

### ISSUE THREAD

**Issue by udit0802** [microsoft/playwright#9253]
Is there a way to execute the tests in distributed mode i.e test distributed accross multiple machines.
I have gone through [1] though, but didn't get how to provide IP/Configuration of the machines to run in distributed mode.
Could someone please answer this query.
Thanks!!

[1]: https://playwright.dev/docs/test-parallel#shard-tests-between-multiple-machines

**Comment by aslushnikov** [comment]
As of today, you can use services like Browserstack that provide this capabilities: https://www.browserstack.com/docs/automate/playwright

**Comment by udit0802** [comment]
Yes but we do have our own infrastructure setup , it is quite difficult to convince people to move to another service for just some tests, it would be pretty nice if we could use the same infrastructure.

**Comment by pavelfeldman** [comment]
What kind of infrastructure do you have in place?

**Comment by udit0802** [comment]
Like for running selenium based tests we have setup selenium hub on openstack machines where multiple tests can be run in parallel, as there is support in selenium.
Do this answer your question?

**Comment by sridharreddysuram123** [comment]
Hi @dgozman - Any update on this? By when we get this feature?

**Comment by muodov** [comment]
Just to make sure we're on the same page, you can [connect Playwright to Selenium 4 hub](https://playwright.dev/docs/selenium-grid). So if you already have a selenium infra, you can probably use it to run playwright scripts

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
